.github/workflows/publish.yml

```diff
@@ -0,0 +1,23 @@
+name: Publish
+
+on:
+  release:
+    types: [ published ]
+
+jobs:
+
+  publish:
+    name: Publish the release version
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout the repository and the branch
+        uses: actions/checkout@v3
+
+      - name: Setup the release version and overwrite the existing major version tag
+        run: |
+          major_version=$(echo $GITHUB_REF_NAME | cut -d. -f1)
+          git config --local user.email "github-actions[bot]@users.noreply.github.com"
```

Comment:
@ZPascal

By the way:  there is also a GitHub Action to take care for exactly this setup of the GitHub Actions bot.  You can call it as follows:

```yaml
      - name: user
        uses: fregante/setup-git-user@v1.1.0
```

It needs to have access to the file contents, i.e. it requires a preceding `actions/checkout` call and at least `permissions.contents: read` in the permissions section.  Also large organisations like Adobe and Ansible make use of this Action; see https://github.com/fregante/setup-git-user/network/dependents (page 2, as of this writing) for reference.

Maybe this could help to enhance the maintainability of this workflow.

Reply:
@kevinmatthes Thanks for the hint. I'll check that.